### PR TITLE
Expose test fakes as a package

### PR DESCRIPTION
**Pull request by yiannistri**
**What changed?**
- Exposed test fakes as a separate package
- Added missing interface checks

**Why?**
To allow use of fake objects from other projects that import Weave GitOps

**How did you test it?**
No new tests, mostly moved code around

**Release notes**
N/A

**Documentation Changes**
N/A

### CODE REVIEW

File: pkg/fakes/fakes.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package fakes
```
Comment (foot) [inline pkg/fakes/fakes.go:1]: The pattern I have seen around is to nest the fakes under the main implementation, could that work or might introduce cyclic deps?

File: pkg/server/config_gettter_test.go [hunk 2]
```diff
@@ -5,12 +5,13 @@ import (
 	"testing"
```
Comment (foot) [inline pkg/server/config_gettter_test.go:5]:
```patch
- pkg/server/config_gettter_test.go
+ pkg/server/config_getter_test.go
```

File: pkg/fakes/fakes.go (hunk 1 above)
Comment (yiannistri) [reply pkg/fakes/fakes.go:1]: Let me try that 👍

Comment (yiannistri) [reply pkg/fakes/fakes.go]: Works fine 👌

**Review by foot — APPROVED** [review]
All looks good to me! 👍  Nice syntactic cleanup w/ the `NewDefaultKubeGetter`

**Pull request merged**